File: app/src/main/java/com/apicloud/pkg/sdk/mvp/presenter/MePresenter.java
package com.apicloud.pkg.sdk.mvp.presenter;


import android.content.Context;

import com.apicloud.pkg.sdk.api.AppApi;
import com.apicloud.pkg.sdk.common.RxPresenter;
import com.apicloud.pkg.sdk.mvp.contract.MeContract;

import javax.inject.Inject;

public class MePresenter extends RxPresenter<MeContract.View> implements MeContract.Presenter {

    private AppApi appApi;
    private Context context;

    @Inject
    public MePresenter(AppApi appApi, Context context) {
        super();
        this.appApi = appApi;
        this.context = context;
    }

}

File: youxian/src/main/java/apicloud/com/youxian/mvp/presenter/MePresenter.kt
package apicloud.com.youxian.mvp.presenter

import android.app.Application
import apicloud.com.youxian.mvp.contract.MeContract
import com.jess.arms.di.scope.ActivityScope
import com.jess.arms.http.imageloader.ImageLoader
import com.jess.arms.integration.AppManager
import com.jess.arms.mvp.BasePresenter
import io.reactivex.Observable
import io.reactivex.Observer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.functions.Action
import io.reactivex.schedulers.Schedulers
import me.jessyan.rxerrorhandler.core.RxErrorHandler
import java.util.concurrent.TimeUnit
import javax.inject.Inject


@ActivityScope
class MePresenter @Inject
constructor(model: MeContract.Model, rootView: MeContract.View, private var mErrorHandler: RxErrorHandler?, private var mApplication: Application?, private var mImageLoader: ImageLoader?, private var mAppManager: AppManager?) : BasePresenter<MeContract.Model, MeContract.View>(model, rootView) {

    override fun onDestroy() {
        super.onDestroy()
        this.mErrorHandler = null
        this.mAppManager = null
        this.mImageLoader = null
        this.mApplication = null
    }

    fun showDatas() {
        Observable.timer(5000L, TimeUnit.MILLISECONDS)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe()
        mRootView.hideLoading()
    }
}
